### PR TITLE
Batch actions

### PR DESCRIPTION
**Pull request by matt-riemer**
accounts___video_tax_news

### CODE REVIEW

**Comment by jfeaver** [comment]
:+1: Nice blurring!

File: README.md [hunk 1]
```diff
@@ -435,6 +435,59 @@ Optionally uses the authorization method (below) to determine if the `current_us
...
+```
+
+You still need to write your own controller action to process the bulk action.  Something like:
+
+```ruby
+class PostsController < ApplicationController
+  def bulk_approve
+    @posts = Post.where(id: params[:ids])
+
+    # You should probably write this inside a transaction.  This code is just for example.
```
Comment (jfeaver) [inline README.md]: This is just an example.

File: app/assets/javascripts/effective_datatables/bulk_actions.js.coffee [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ).done((response) ->
+    success = response['message'] || "Successfully completed #{title} bulk action"
+    $table.siblings('.dataTables_processing').html(success)
+  ).fail((response) ->
+    error = response['message'] || "An error occured while attempting #{title} bulk action: #{response.statusText}"
+    alert(error)
+  ).always((response) ->
+    $table.dataTable().data('bulk-actions-restore-selected-values', values)
+    $table.DataTable().draw()
+  )
```
Comment (jfeaver) [inline app/assets/javascripts/effective_datatables/bulk_actions.js.coffee]: This is some good looking js!

File: app/models/effective/effective_datatable/rendering.rb [hunk 3]
```diff
@@ -71,7 +71,8 @@ def arrayize(collection)
               edit_action: (opts[:partial_locals] || {})[:edit_action],
               destroy_action: (opts[:partial_locals] || {})[:destroy_action],
               unarchive_action: (opts[:partial_locals] || {})[:unarchive_action]
-            }
+            )
```
Comment (jfeaver) [inline app/models/effective/effective_datatable/rendering.rb]: It'd be nice if Hashes were indifferent by default. With the new string optimizations, maybe this is being considered??

File: app/views/effective/datatables/_bulk_actions_dropdown.html.haml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.btn-group.buttons-bulk-actions
+  %button.btn.btn-default.dropdown-toggle{'type' => 'button', 'data-toggle' => 'dropdown', 'aria-haspopup' => true, 'aria-expanded' => false, 'disabled' => 'disabled'}
+    Bulk Actions
+    %span.caret
+  %ul.dropdown-menu
+    - if content_for?(:effective_datatables_bulk_actions)
+      = yield :effective_datatables_bulk_actions
+    - else
+      %li
+        %a{href: '#'} No bulk actions
```
Comment (jfeaver) [inline app/views/effective/datatables/_bulk_actions_dropdown.html.haml]: Why is this a link?

File: app/models/effective/effective_datatable/rendering.rb (hunk 3 above)
Comment (matt-riemer) [inline app/models/effective/effective_datatable/rendering.rb]: In practice, that would be so handy!

File: app/views/effective/datatables/_bulk_actions_dropdown.html.haml (hunk 4 above)
Comment (matt-riemer) [inline app/views/effective/datatables/_bulk_actions_dropdown.html.haml]: as per bootstrap html for a dropdown menu

**Pull request merged**